File: study-project-frontend/src/types/account.ts
// 用户数据类型，和后端 Account.java 对应
export interface Account {
  accountId: string   // Long → String，避免 JS 精度丢失
  accountEmail: string
  accountUsername: string
  accountPassword: string
  accountState: number
}

// 状态标签映射
export function getStateTag(state: number) {
  switch (state) {
    case 1: return { text: '正常', type: 'success' as const }
    case 0: return { text: '禁用', type: 'danger' as const }
    default: return { text: '未知', type: 'info' as const }
  }
}

File: study-project-backend/src/main/java/com/shankong/pojo/Account.java
package com.shankong.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("db_account")  // 指定对应的数据库表名
public class Account implements Serializable {
    @Serial
    private static final long serialVersionUID = -8940196742313994740L;

    @TableId(value = "account_id", type = IdType.ASSIGN_ID)
    @JsonSerialize(using = ToStringSerializer.class)
    private Long accountId;

    @TableField("account_email")
    private String accountEmail;

    @TableField("account_username")
    private String accountUsername;

    @TableField("account_password")
    private String accountPassword;

    @TableField("account_state")
    private Integer accountState;
}
